File: ms-clientes/src/main/java/com/smartlogix/msclientes/model/Cliente.java
package com.smartlogix.msclientes.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "cliente")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_cliente")
    private Long idCliente;

    @Column(name = "id_usuario_auth", unique = true)
    private Long idUsuarioAuth;

    @Column(name = "rut", unique = true)
    private String rut;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "apellido_paterno")
    private String apellidoPaterno;

    @Column(name = "apellido_materno")
    private String apellidoMaterno;

    @Column(name = "correo", unique = true)
    private String correo;

    @Column(name = "telefono")
    private String telefono;

    @OneToMany(mappedBy = "cliente", cascade = CascadeType.ALL, orphanRemoval = true)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    @JsonManagedReference
    @Builder.Default
    private List<DireccionCliente> direcciones = new ArrayList<>();
}

File: frontend/src/features/clientes/types/cliente.ts
export interface DireccionCliente {
  idDireccion?: number;
  idCliente?: number;
  idComuna?: number;
  calle: string;
  numero: string;
  detalle?: string;
  esPrincipal?: boolean;
}

export interface DireccionPrincipalRequest {
  idComuna: number;
  calle: string;
  numero: string;
  detalle?: string;
}

export interface Cliente {
  idCliente?: number;
  idUsuarioAuth?: number;
  rut: string;
  nombre: string;
  apellidoPaterno: string;
  apellidoMaterno: string;
  correo: string;
  telefono: string;
  region?: string;
  direcciones?: DireccionCliente[];
  direccionPrincipal?: DireccionCliente | null;
}

export interface CrearClienteRequest {
  idUsuarioAuth?: number;
  rut: string;
  nombre: string;
  apellidoPaterno: string;
  apellidoMaterno: string;
  correo: string;
  telefono: string;
  direccionPrincipal: DireccionPrincipalRequest;
}